core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -106,4 +123,73 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  /**
+   * Traverses the history of the table's current snapshot and finds the snapshot with the given snapshot id as its
+   * parent.
+   * @return the snapshot for which the given snapshot is the parent
+   * @throws IllegalArgumentException when the given snapshotId is not found in the table
+   * @throws IllegalStateException when the given snapshotId is not an ancestor of the current table state
+   */
+  public static Snapshot snapshotAfter(Table table, long snapshotId) {
+    Preconditions.checkArgument(table.snapshot(snapshotId) != null, "Cannot find parent snapshot: %s", snapshotId);
+
+    Snapshot current = table.currentSnapshot();
+    while (current != null) {
+      if (current.parentId() == snapshotId) {
+        return current;
+      }
+
+      current = table.snapshot(current.parentId());
+    }
+
+    throw new IllegalStateException(
+        String.format("Cannot find snapshot after %s: not an ancestor of table's current snapshot", snapshotId));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: snapshotId (declared at line 134). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 147. Consider formatting this data with &#39;%d&#39; instead.